meter/lgpcs/lgpcs.go

```diff
@@ -120,53 +122,99 @@ func (m *Com) Login() error {
 }
 
 // Data gives the data read from the pcs.
-func (m *Com) Data() (EssData8, error) {
+func (m *Com) Data() (EssData, error) {
 	return m.dataG()
 }
 
 // refreshData reads data from lgess pcs. Tries to re-login if "405" auth_key expired is returned
-func (m *Com) refreshData() (EssData8, error) {
+func (m *Com) refreshData() (EssData, error) {
+	var data EssData
+	var err error
+	if m.essType == LgEss8 {
+		var meterResponse MeterResponse8
+		if err = refreshDataFromLgess[MeterResponse8](&meterResponse, m); err != nil {
+			return EssData{}, err
+		}
+		essDataFromMeterResponse8(&data, &meterResponse)
+	} else {
+		var meterResponse MeterResponse15
+		if err = refreshDataFromLgess[MeterResponse15](&meterResponse, m); err != nil {
+			return EssData{}, err
+		}
+		essDataFromMeterResponse15(&data, &meterResponse)
+	}
+
+	return data, nil
+}
+
+// read data from Lgess, reconnects if session expired
+func refreshDataFromLgess[T MeterResponse8 | MeterResponse15](meterResponse *T, m *Com) error {
```

Comment:
How about

    func (m *Com) refresh(res any) error 

and then call it with

    var res EssData8
    err := m.refresh(&res)